frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -30,10 +30,25 @@ frappe.ui.form.on("System Settings", {
 			frm.set_value("bypass_restrict_ip_check_if_2fa_enabled", 0);
 		}
 	},
-	on_update: function (frm) {
-		if (frappe.boot.time_zone && frappe.boot.time_zone.system !== frm.doc.time_zone) {
-			// Clear cache after saving to refresh the values of boot.
-			frappe.ui.toolbar.clear_cache();
+	after_save: function (frm) {
+		/**
+		 * Checks whether the effective value has changed.
+		 *
+		 * @param {Array.<string>} - Tuple with new fallback, previous fallback and
+		 *   optionally an override value.
+		 * @returns {boolean} - Whether the resulting value has effectively changed
+		 */
+		const has_effectively_changed = ([new_fallback, prev_fallback, override = undefined]) =>
+			!override && prev_fallback !== new_fallback;
+
+		const attr_tuples = [
+			[frm.doc.language, frappe.boot.sysdefaults.language, frappe.boot.user.language],
+			[frm.doc.rounding_method, frappe.boot.sysdefaults.rounding_method], // no user override.
+		];
+
+		if (attr_tuples.some(has_effectively_changed)) {
+			frappe.msgprint(__("Refreshing..."));
+			window.location.reload();
```

Comment:
Shouldn't this ideally be `frappe.ui.toolbar.clear_cache();` instead of `window.location.reload();`?
@bosue @ankush

Reply:
@rtdany10: It wasn‘t before, but maybe it should, yes.
We need to know for sure if it is necessary, then do a followup.